File: backend/app/routers/admin/inquiries.py
from fastapi import APIRouter, Depends, HTTPException
from bson import ObjectId
from app.database import get_database
from app.middleware.auth import require_admin
from app.schemas.inquiry import UpdateInquiryRequest, InquiryResponse

router = APIRouter()


@router.get("")
async def list_inquiries(status: str | None = None, admin: dict = Depends(require_admin)):
    db = get_database()
    query = {}
    if status:
        query["status"] = status
    cursor = db.inquiries.find(query).sort("created_at", -1)
    items = []
    async for inq in cursor:
        items.append(InquiryResponse(
            id=str(inq["_id"]),
            name=inq["name"],
            email=inq["email"],
            phone=inq.get("phone"),
            package_id=inq.get("package_id"),
            message=inq["message"],
            event_date=inq.get("event_date"),
            event_time=inq.get("event_time"),
            event_duration=inq.get("event_duration"),
            status=inq["status"],
            created_at=inq["created_at"],
        ))
    return {"inquiries": items}


@router.put("/{inquiry_id}")
async def update_inquiry(inquiry_id: str, body: UpdateInquiryRequest, admin: dict = Depends(require_admin)):
    db = get_database()
    valid_statuses = ["new", "contacted", "booked", "closed"]
    if body.status not in valid_statuses:
        raise HTTPException(status_code=400, detail=f"Status must be one of: {valid_statuses}")
    result = await db.inquiries.update_one(
        {"_id": ObjectId(inquiry_id)},
        {"$set": {"status": body.status}},
    )
    if result.matched_count == 0:
        raise HTTPException(status_code=404, detail="Inquiry not found")
    return {"message": "Updated"}

File: backend/src/routes/admin/inquiries.ts
import { Router, Response } from 'express';
import { ObjectId } from 'mongodb';
import { getDb } from '../../database';
import { requireAdmin, AuthRequest } from '../../middleware/auth';

const router = Router();

router.get('/', requireAdmin, async (req: AuthRequest, res: Response): Promise<void> => {
  const db = await getDb();
  const query: Record<string, unknown> = {};
  if (req.query.status) query.status = req.query.status;
  const inquiries = await db.collection('inquiries').find(query).sort({ created_at: -1 }).toArray();
  res.json({ inquiries: inquiries.map((inq) => ({ id: inq._id.toString(), name: inq.name, email: inq.email, phone: inq.phone ?? null, package_id: inq.package_id ?? null, message: inq.message, event_date: inq.event_date ?? null, event_time: inq.event_time ?? null, event_duration: inq.event_duration ?? null, status: inq.status, created_at: inq.created_at })) });
});

router.get('/:inquiryId', requireAdmin, async (req: AuthRequest, res: Response): Promise<void> => {
  const db = await getDb();
  const inq = await db.collection('inquiries').findOne({ _id: new ObjectId(req.params.inquiryId) });
  if (!inq) { res.status(404).json({ detail: 'Inquiry not found' }); return; }
  res.json({ id: inq._id.toString(), name: inq.name, email: inq.email, phone: inq.phone ?? null, package_id: inq.package_id ?? null, message: inq.message, event_date: inq.event_date ?? null, event_time: inq.event_time ?? null, event_duration: inq.event_duration ?? null, status: inq.status, created_at: inq.created_at });
});

router.put('/:inquiryId', requireAdmin, async (req: AuthRequest, res: Response): Promise<void> => {
  const valid = ['new', 'contacted', 'booked', 'closed'];
  if (!valid.includes(req.body.status)) { res.status(400).json({ detail: `Status must be one of: ${valid.join(', ')}` }); return; }
  const db = await getDb();
  const result = await db.collection('inquiries').updateOne({ _id: new ObjectId(req.params.inquiryId) }, { $set: { status: req.body.status } });
  if (!result.matchedCount) { res.status(404).json({ detail: 'Inquiry not found' }); return; }
  res.json({ message: 'Updated' });
});

export default router;
